_data-prepper/core-apis.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: default
+title: Core APIs
+nav_order: 2
+---
+
+# Core APIs
+
+All Data Prepper instances expose a server with some control APIs. By default, this server runs on port 4900. Some plugins, especially source plugins may expose other servers. These will be on different ports, and their configurations are independent of the core API. For example, to shut down Data Prepper, you can run the following curl request:
+
+```
+curl -X POST http://localhost:4900/shutdown
+```
+
+## APIs
+
+The following table lists the available APIs.
+
+| Name | Description |
+| --- | --- | 
+| ```GET /list```<br>```POST /list``` | Returns a list of running pipelines. |
+| ```POST /shutdown``` | Starts a graceful shutdown of Data Prepper. |
+| ```GET /metrics/prometheus```<br>```POST /metrics/prometheus``` | Returns a scrape of Data Prepper metrics in Prometheus text format. This API is available as a `metricsRegistries` parameter in the Data Prepper configuration file `data-prepper-config.yaml` and contains `Prometheus` as part of the registry.
+| ```GET /metrics/sys```<br>```POST /metrics/sys``` | Returns JVM metrics in Prometheus text format. This API is available as a `metricsRegistries` parameter in the Data Prepper configuration file `data-prepper-config.yaml` and contains `Prometheus` as part of the registry.
+
+## Configuring the server
+
+You can configure your Data Prepper core APIs through the `data-prepper-config.yaml` file. 
+
+### SSL/TLS connection
+
+Many of the Getting Started guides for this project disable SSL on the endpoint:
+
+```yaml
+ssl: false
+```
+
+To enable SSL on your Data Prepper endpoint, configure your `data-prepper-config.yaml` file with the following options:
+
+```yaml
+ssl: true
+keyStoreFilePath: "/usr/share/data-prepper/keystore.p12"
+keyStorePassword: "secret"
+privateKeyPassword: "secret"
+```
+
+For more information about configuring your Data Prepper server with SSL, see [Server Configuration](https://github.com/opensearch-project/data-prepper/blob/main/docs/configuration.md#server-configuration). If you are using a self-signed certificate, you can add the `-k` flag to quickly test sending curl requests for the core APIs with SSL. Use the following curl request to test for core APIs with SSL.
```

Comment:
Maybe delete "sending". We know the request is doing that.
"... you can add the `-k` flag to the request to quickly test core APIs with SSL."

Is the "shutdown" command (in example) used intentionally to test curl requests? If yes, can you just state explicitly, "Use the `shutdown` request to test core APIs using SSL."?
Or can you use any of the core APIs (## APIs) to test?

Reply:
@oeyh Can we say, "Use the `shutdown` request to test core APIs using SSL."?

Reply:
Sure.